File: app/src/main/java/com/mxt/anitrend/data/converter/UserOptionsConverter.java
package com.mxt.anitrend.data.converter;

import com.mxt.anitrend.model.api.retro.WebFactory;
import com.mxt.anitrend.model.entity.anilist.meta.UserOptions;

import io.objectbox.converter.PropertyConverter;

/**
 * Created by max on 2018/03/22.
 */

public class UserOptionsConverter implements PropertyConverter<UserOptions, String> {

    @Override
    public UserOptions convertToEntityProperty(String databaseValue) {
        if(databaseValue == null)
            return null;
        return WebFactory.gson.fromJson(databaseValue, UserOptions.class);
    }

    @Override
    public String convertToDatabaseValue(UserOptions entityProperty) {
        if(entityProperty == null)
            return null;
        return WebFactory.gson.toJson(entityProperty);
    }
}

File: app/src/main/java/com/mxt/anitrend/data/converter/UserOptionsConverter.kt
package com.mxt.anitrend.data.converter

import com.mxt.anitrend.model.api.retro.ServiceFactory
import com.mxt.anitrend.model.entity.anilist.meta.UserOptions
import io.objectbox.converter.PropertyConverter

/**
 * Created by max on 2018/03/22.
 */
class UserOptionsConverter : PropertyConverter<UserOptions, String> {
    override fun convertToEntityProperty(databaseValue: String?): UserOptions? = databaseValue?.let { ServiceFactory.gson.fromJson(it, UserOptions::class.java) }

    override fun convertToDatabaseValue(entityProperty: UserOptions?): String? = entityProperty?.let { ServiceFactory.gson.toJson(it) }
}
